node/network/availability-recovery/src/lib.rs

```diff
@@ -1106,10 +1193,67 @@ impl AvailabilityRecoverySubsystem {
 		let mut state = State::default();
 		let Self { recovery_strategy, mut req_receiver, metrics } = self;
 
+		let (erasure_task_tx, erasure_task_rx) = futures::channel::mpsc::channel(16);
+		let mut erasure_task_rx = erasure_task_rx.fuse();
+
 		loop {
 			let recv_req = req_receiver.recv(|| vec![COST_INVALID_REQUEST]).fuse();
 			pin_mut!(recv_req);
 			futures::select! {
+				erasure_task = erasure_task_rx.next() => {
+					match erasure_task {
+						Some(ErasureTask::Reconstruct(n_validators, chunks, sender )) => {
+							let result = ctx.spawn_blocking("reconstruct_v1", Box::pin(async move {
```

Comment:
Would it make sense to have 1 or 2 permanent threads or a pool that receive and process messages instead of spawning temporary threads?

Reply:
AFAIK tokio already has some threads that it reuses. If we would have many parallel tasks then yeah I'd say it makes sense to limit them to some fixed number. This is not really the case for normal operation.

Reply:
While tokio has it's own thread pool it does not mean that this pool has any available threads for reusing. In general, having a persistent threads or event a persistent thread pool for some frequent and CPU intensive task is a good idea. I personally prefer processes to threads in this case, but it is not applicable here apparently.

Reply:
After sassafras, we'd expect k = relayVrfModuloSamples + one std deviation threads here every six seconds, so maybe 5 or 6 new threads per second, or use k to size the thread pool.

Reply:
It would make sense to have a thread pool for intensive CPU tasks but for the whole system not just one subsystem. I’ll look into using rayon for this. I preferred spawn blocking for simplicity and not really making a noticrable difference in performance whilr still fixing a major issue of blocking a cpu for that long.

Reply:
> After sassafras, we'd expect k = relayVrfModuloSamples + one std deviation threads here every six seconds, so maybe 5 or 6 new threads per second, or use k to size the thread pool.

It would only need 5 or 6 if we expect av recovery to conclude at the same time for all candidates and the size of the PoVs to be big.  I think we can get away with way less, since even in the worst case it would take 500-700ms to complete reconstruct and re-encoding, so we could serialize some of the work and still finish within 3s.

Reply:
Implemented a pool of 2 workers.